tests/cli/wit-deep-variant.wit

```diff
@@ -0,0 +1,47 @@
+// FAIL[parse]: component wit % -t
+
+// TODO: this should use "FAIL" and should actually be a test, but this
+// currently takes too long to run and the long runtime is in and of itself a
+// bug.
+// fail[abi]: component embed --dummy % | component new
```

Comment:
This is an interesting test where the test below actually fits in the flat representation, unlike all of the above tests which don't fit flat. This test takes quite a long time to produce the flat representation.

This isn't a problem for validation since this type is "too large and complex" and fails validation before we get to the flat representation, but is still an issue for tooling which I'm not sure how best to handle.